### PR TITLE
Added the  AI-Powered Super-Resolution and Image Restoration model

### PR DESCRIPTION
**Pull request by Peart-Guy**
### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
AI-Powered Super-Resolution and Image Restoration for Historical Photos

- [x] I have provided the issue title.

---

#### Info about the Related Issue
**What's the goal of the project?**  

This project aims to create an AI-Powered Image Restoration Tool that transforms low-quality historical photos by upscaling, denoising, and adding realistic color. Using deep learning, it revives old images with stunning clarity and detail, preserving memories in a vibrant, lifelike form. Perfect for heritage conservation and digital archiving!

- [x] I have described the aim of the project.

---

#### Name

Ankan

- [x] I have provided my name.

---

#### GitHub ID

Peart-Guy

- [x] I have provided my GitHub ID.

---

#### Email ID
**Please mention your email ID for further communication.**  
*Enter your email ID here.*

- [x] I have provided my email ID.

---

#### Identify Yourself

GSSOC-24 EXTD

- [x] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #1080 *

- [x] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
I have added the model

- [x] I have described my changes.

---

#### Type of Change
**Select the type of change:**  
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

---

#### How Has This Been Tested?
**Describe how your changes have been tested.**  
*Describe your testing process here.*

- [x] I have described my testing process.

---

#### Checklist
**Please confirm the following:**  
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by Peart-Guy** [comment]
@UTSAVS26  Merge the PR

**Comment by UTSAVS26** [comment]
> @UTSAVS26  Merge the PR

Wait for sometime mentors are reviewing.

**Comment by Peart-Guy** [comment]
> > @UTSAVS26  Merge the PR
> 
> Wait for sometime mentors are reviewing.

Ok, But Kindly make it fast since it is the  end days of GSSOC-24 EXTD

File: Image processing/AI Super-Resolution and Image Restoration/main.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    def forward(self, x):
+        x = torch.nn.functional.relu(self.conv1(x))
+        x = torch.nn.functional.relu(self.conv2(x))
+        x = self.conv3(x)
+        return x
+
+# Initialize the model and load weights if available
+model = SRCNN()
+model.load_state_dict(torch.load('srcnn_pretrained.pth', map_location='cpu'))
```
Comment (ruhi47) [inline Image processing/AI Super-Resolution and Image Restoration/main.py:24]: Hey @Peart-Guy ,
Either provide the link to download the datasets into the README file or put a folder with datasets.
Update this and then I will review again.

**Review by ruhi47 — CHANGES_REQUESTED** [review]

**Comment by TheChaoticor** [comment]
@Peart-Guy , please complete it within tomorrow with the necessary changes

**Comment by Peart-Guy** [comment]
@TheChaoticor @UTSAVS26 @ruhi47 
I have made the changes by adding the  datasets model in the readme file. Kindly check it out

**Pull request merged**